scio-test/src/main/scala/com/spotify/scio/testing/SCollectionMatchers.scala

```diff
@@ -29,18 +29,149 @@ import org.apache.beam.sdk.transforms.SerializableFunction
+   * but will but have a nicer message in case of failure.
+   */
+  def containsInAnyOrder[T: Coder](
+    ts: Seq[T],
+    context: ScioContext
+  ): h.Matcher[JIterable[T]] =
+    SerializableMatchers.fromSupplier {
+      supplierFromCoder(ts, context) { ds =>
+        val items = ds.mkString("\n\t\t", "\n\t\t", "\n")
+        val message = s"Expected: iterable with items [$items]"
+        val c = Matchers
+          .containsInAnyOrder(ds: _*)
+          .asInstanceOf[h.Matcher[JIterable[T]]]
+        Matchers.describedAs(message, c)
+      }
+    }
+
+  def makeFn[T](
+    f: JIterable[T] => Unit
+  ): SerializableFunction[JIterable[T], Void] =
+    new SerializableFunction[JIterable[T], Void] {
+      // delegate serialization to Kryo to avoid serialization issues in tests
+      // when a non-serializable object is captured by the closure
+      private[this] val impl = Externalizer(f)
+
+      override def apply(input: JIterable[T]): Void = {
+        impl.get(input)
+        null
+      }
+    }
+
+  def makeFnSingle[T](f: T => Unit): SerializableFunction[T, Void] =
+    new SerializableFunction[T, Void] {
+      // delegate serialization to Kryo to avoid serialization issues in tests
+      // when a non-serializable object is captured by the closure
+      private[this] val impl = Externalizer(f)
+
+      override def apply(input: T): Void = {
+        impl.get(input)
+        null
+      }
+    }
+
+  def assertThatFn[T: Eq: Coder](mm: h.Matcher[JIterable[TestWrapper[T]]]) =
+    makeFn[T](in => assertThat(TestWrapper.wrap(in), mm))
+
+  def assertThatNotFn[T: Eq: Coder](mm: h.Matcher[JIterable[TestWrapper[T]]]) =
+    makeFn[T](in => assertThat(TestWrapper.wrap(in), Matchers.not(mm)))
+
+  def assert[T: Eq: Coder](p: Iterable[TestWrapper[T]] => Boolean) =
+    makeFn[T](in => Predef.assert(p(TestWrapper.wrap(in).asScala)))
+
+  def assertSingle[T: Eq: Coder](p: TestWrapper[T] => Boolean) =
+    makeFnSingle[T](in => Predef.assert(p(TestWrapper(in))))
+
+  def assertNot[T: Eq: Coder](p: Iterable[TestWrapper[T]] => Boolean) =
+    makeFn[T](in => Predef.assert(!p(TestWrapper.wrap(in).asScala)))
+
+  def assertNotSingle[T: Eq: Coder](p: TestWrapper[T] => Boolean) =
+    makeFnSingle[T](in => Predef.assert(!p(TestWrapper(in))))
+
+  def isEqualTo[T: Eq: Coder](t: T): SerializableFunction[T, Void] = {
+    val mm = Matchers.equalTo[Any](TestWrapper(t))
+    makeFnSingle[T](in => assertThat(TestWrapper(in), mm))
+  }
+
+  def notEqualTo[T: Eq: Coder](t: T): SerializableFunction[T, Void] = {
+    val mm = Matchers.not(Matchers.equalTo[Any](TestWrapper(t)))
+    makeFnSingle[T](in => assertThat(TestWrapper(in), mm))
+  }
+}
 
 /**
  * Trait with ScalaTest [[org.scalatest.matchers.Matcher Matcher]]s for
  * [[com.spotify.scio.values.SCollection SCollection]]s.
  */
-trait SCollectionMatchers {
+trait SCollectionMatchers extends EqInstances {
+
+  import ScioMatchers.{makeFn, makeFnSingle}
```

Comment:
```suggestion
  import ScioMatchers.makeFn
```

Reply:
The other one is not used ... either this or replace `makeFn` with `ScioMatchers.makeFn` up to you 😄